Cargo.toml

```diff
@@ -1,7 +1,9 @@
 [workspace]
 members = [
     "rust/candid",
+    "rust/candid_parser",
     "rust/candid_derive",
+    "rust/ic_principal",
```

Comment:
Do you recall why we put `Principal` back to `candid` a while ago? Also do we want to give it a more general name, e.g. `ic_types`, to accommodate more types in the future?

Reply:
We used to have the [`ic-types` crate](https://crates.io/crates/ic-types) which contained `principal` and `hash_tree` modules.
The modules were moved to `candid` and `agent-rs` repo respectively. And `ic-types` is [deprecated](https://github.com/dfinity/ic-types/pull/27).

IMO, it's better to have `ic_principal` crate which contains only `Principal`. It will be a stable crate on the top of the dependency tree. We won't have to bump its version because of changes in other parts.